diagnostics/docker-worker/taskCompleted.js

```diff
@@ -0,0 +1,126 @@
+    ).then(function(){
+      debug("Submitting task: %s", taskId);
+
+      return helper.queue.createTask(taskId, {
+        provisionerId:    'aws-provisioner',
+        workerType:       'v2',
+        created:          taskcluster.utils.fromNow(),
+        deadline:         taskcluster.utils.fromNow('1 hour'),
+        payload:          {
+          image:          "ubuntu:14.04",
+          command:        ["/bin/bash", "-c", "exit $MY_ENV_VAR;"],
+          env: {
+            MY_ENV_VAR:   '0'
+          },
+          maxRunTime:     20 * 60
+        },
+        metadata: {
+          name:           "Completed task test",
+          description:    "Task for docker-worker that tests environment " +
+                          "variables and completes successfully",
+          owner:          "nobody@localhost.local",
+          source:         "https://github.com/taskcluster/taskcluster-diagnostics"
+        }
+      });
+    }).then(function(){
+      return helper.receiver.listenFor(
+        'taskPending',
+        helper.queueEvents.taskPending({taskId: taskId}));
+    }).then(function(){
+      debug("Scheduling task");
+      return helper.queue.scheduleTask(taskId); //note, here we don't pass an object?! This isn't clear from the taskcluster-client docs
+    }).then(function(){
+      return helper.receiver.listenFor(
+        'taskRunning',
+        helper.queueEvents.taskRunning({taskId: taskId}));
+    }).then(function(){
+      debug("Claiming task");
+      return helper.queue.claimTask(taskId, 0, {
+        workerGroup: "test-worker-group",
+        workerId: "test-worker-id"
+      });
+    }).then(function(){
+      return helper.receiver.listenFor(
+        'taskCompleted',
+        helper.queueEvents.taskCompleted({taskId: taskId}));
+    }).then(function(){
+      /*
+      In this phase of the test we create an artifact and then retrieve it
+      and assert that retrieved === expected. To do this we:
+      1) ask the Queue to provide us with a URL we can PUT to, by calling
+         queue.createArtifact.
+      2) Then we submit a PUT with the test artifact to the url we were give.
+      3) We use queue.buildURL to get the URL at which the test artifact is
+         available, do a GET and compare the received artifact to the
+         expected artifact.
+      */
+      var artifactOptions = {
+        "storageType": "s3",
+        "expires": taskcluster.utils.fromNow("1 hour"),
+        "contentType": "application/json",
+      };
+      return helper.queue.createArtifact(taskId, runID, testArtifactName, artifactOptions);
+    }).then(function(queueResponse){
+      // 2) We've received a response from the Queue, it should have a PUT URL in it.
+      debug("Creating artifact");
+      if (queueResponse.putUrl) {
+        request.put(queueResponse.putUrl)
+        .set('Content-Type', 'application/json')
+        .set("Content-Length", testArtifact.length)
+        .send(testArtifact).end();
```

Comment:
Hmm... You should probably wait for the request to be done, before you proceed..
If used superagent-promise, then just:
```js
return request.put(queueResponse.putUrl)
            .set('Content-Type', 'application/json')
            .set("Content-Length", testArtifact.length)
            .send(testArtifact).end();
```

Should do the trick... Otherwise you risk that you try to download the artifact before upload was completed. Naturally that won't work..